deploy.go

```diff
@@ -197,43 +201,27 @@ func (d *imageDeployer) computePCRProtectionProfile(esp string, env secboot_efi.
 		kernelPaths = append(kernelPaths, filepath.Join(esp, "EFI/ubuntu/grubx64.efi"))
 	}
 
-	var kernels []*secboot_efi.ImageLoadEvent
+	var kernels []secboot_efi.ImageLoadActivity
 	for _, path := range kernelPaths {
 		log.Debugln("found kernel", path)
-		kernels = append(kernels, &secboot_efi.ImageLoadEvent{
-			Source: secboot_efi.Shim,
-			Image:  secboot_efi.FileImage(path)})
-	}
-
-	loadSequences := &secboot_efi.ImageLoadEvent{
-		Source: secboot_efi.Firmware,
-		Image:  secboot_efi.FileImage(filepath.Join(esp, "EFI/ubuntu/shimx64.efi")),
-		Next:   kernels}
-
-	if d.opts.AddEFIBootManagerProfile {
-		log.Debugln("adding boot manager PCR profile")
-		params := secboot_efi.BootManagerProfileParams{
-			PCRAlgorithm:  tpm2.HashAlgorithmSHA256,
-			LoadSequences: []*secboot_efi.ImageLoadEvent{loadSequences},
-			Environment:   env}
-		if err := secboot_efi.AddBootManagerProfile(pcrProfile, &params); err != nil {
-			return nil, fmt.Errorf("cannot add EFI boot manager profile: %w", err)
-		}
+		kernels = append(kernels, secboot_efi.NewImageLoadActivity(secboot_efi.FileImage(path)))
 	}
 
-	if d.opts.AddEFISecureBootProfile {
-		log.Debugln("adding secure boot policy PCR profile")
-		params := secboot_efi.SecureBootPolicyProfileParams{
-			PCRAlgorithm:  tpm2.HashAlgorithmSHA256,
-			LoadSequences: []*secboot_efi.ImageLoadEvent{loadSequences},
-			Environment:   env}
-		if err := secboot_efi.AddSecureBootPolicyProfile(pcrProfile, &params); err != nil {
-			return nil, fmt.Errorf("cannot add EFI secure boot policy profile: %w", err)
-		}
+	loadSeqs := secboot_efi.NewImageLoadSequences()
+	loadSeqs.Append(secboot_efi.NewImageLoadActivity(
+		secboot_efi.FileImage(filepath.Join(esp, "EFI/ubuntu/shimx64.efi"))).Loads(kernels...))
+
+	if err := secboot_efi.AddPCRProfile(
+		tpm2.HashAlgorithmSHA256,
+		pcrProfile.RootBranch(),
+		loadSeqs,
+		options...,
+	); err != nil {
+		return nil, fmt.Errorf("cannot add PCR profile: %w", err)
 	}
 
 	if d.opts.AddUbuntuKernelProfile {
-		pcrProfile.AddPCRValue(tpm2.HashAlgorithmSHA256, 12, make([]byte, tpm2.HashAlgorithmSHA256.Size()))
+		pcrProfile.RootBranch().AddPCRValue(tpm2.HashAlgorithmSHA256, 12, make([]byte, tpm2.HashAlgorithmSHA256.Size()))
```

Comment:
There is a `WithKernelConfigProfile` option that means these measurements don't need to be added manually, but I've just checked and it doesn't work for this case as it expects a snap model to be set. I also noticed that snapd is not using it yet and probably should be :/